backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CodeVersionHandler.java

```diff
@@ -76,6 +76,13 @@ public CodeVersionHandler(Executor executor, FutureJdbi jdbi) {
             maybeSnapshotId -> {
               if (maybeSnapshotId.isPresent()) {
                 if (request.getOverwrite()) {
+                  jdbi.useHandle(
+                      handle ->
+                          handle
+                              .createUpdate(
+                                  "UPDATE experiment_run SET code_version_snapshot_id = null WHERE id=:run_id")
+                              .bind("run_id", request.getId())
+                              .execute());
```

Comment:
we can not delete code version snapshot directly because experimentRun have foreign key reference so first need to remove code_version_snapshot_id from experimentRun and then we can delete snapshot. so fixed it here.

Reply:
Then we need to put this in a transaction. Otherwise we can set it to `null` but fail to add new value. I have no clue how transactions will work while mixing with hibernate.

Reply:
Can we just remove the foreign key instead?

Reply:
Removing foreign key might be affect many other unwanted places so instead of that i have handled this using hibernate transition and used hibernate sql query instead of jdbi sql query so we can used hibernate transaction.